dev/Microsoft.Process.Environment/EnvironmentManager.cpp

```diff
@@ -63,4 +74,125 @@ namespace winrt::Microsoft::ProjectReunion::implementation
     {
         throw hresult_not_implemented();
     }
+
+    StringMap EnvironmentManager::GetProcessEnvironmentVariables()
+    {
+        //Get the pointer to the process block
+        LPWSTR environmentVariablesString = GetEnvironmentStrings();
+        THROW_HR_IF_NULL(E_POINTER, environmentVariablesString);
+
+        // parse the block into a map
+        LPTSTR variable;
+        variable = (LPTSTR)environmentVariablesString;
+
+        StringMap environmentVariables;
+        while (*variable)
+        {
+            std::wstring environmentVariable(variable);
+            size_t locationOfNull = environmentVariable.find_last_of(L'=');
```

Comment:
Why find_last_of rather than find_first_of?

Reply:
Because the first four Environment Variables of this block are
*  =::=::\
*  =C:=C:\Users\dahoehna
* =D:=D:\git\microsoft\ProjectReunion\BuildOutput\Debug\x64\EnvironmentManagerTests
* =ExitCode=00000000

Reply:
According to GetEnvironmentStrings there should only be 1 "=" and that delimits the value and name.

Reply:
What are those?

Are those defined by CMD.EXE?

>According to GetEnvironmentStrings there should only be 1 "=" and that delimits the value and name.
"should"? (if I could do the Spock one eyebrow quirked thing that's what I'd be doing right now)

Don't your examples above contradict that statement?

For more fun try this at a command line

```
C:>set X="1=1!=2==1+1"

C:>echo %x%
"1=1!=2==1+1"
```

Searching for 'last' is a bad idea

Reply:
> Are those defined by CMD.EXE? 
I have no idea where those are coming from.

> Don't your examples above contradict that statement?
It does.

I'll get it changed.